coresdk/src/coresdk/particle_system.h

```diff
@@ -0,0 +1,104 @@
+#ifndef PARTICLE_SYSTEM_H
+#define PARTICLE_SYSTEM_H
+
+#include "splashkit.h"
+
+#define MAX_PARTICLES 1000 // Maximum number of particles
+#define STREAK_HISTORY 20 // Number of positions to track for the streak
+#define PI 3.14 // setting the PI value
+
+// Particle structure to represent individual particle properties
+struct Particle
+{
+    float x, y;           // Position
+    float dx, dy;         // Velocity
+    float lifespan;       // Remaining lifespan
+    float size;           // Particle size
+    color particle_color; // Particle color
+    bool is_active;       // Active state of the particle
+    bool glow_enabled;    // Whether this particle glows
+    float weight;         // Particle weight
+
+
+};
+
+// Particle system to manage all particles
+struct ParticleSystem
+{
+    Particle particles[MAX_PARTICLES]; // Array of particles
```

Comment:
Is there always going to be the maximum number of particles? If not, std::vector could save some memory.